### PR TITLE
[SPARK-10276][MLlib][PySpark] Add @since annotation to pyspark.mllib.recommendation

### PR DESCRIPTION
**Pull request by yu-iskw**

### CODE REVIEW

File: python/pyspark/mllib/recommendation.py [hunk 1]
```diff
@@ -157,17 +167,25 @@ def recommendProducts(self, user, num):
         return list(self.call("recommendProducts", user, num))
 
     @property
+    @since("1.3.1")
```
Comment (noel-smith) [inline python/pyspark/mllib/recommendation.py]: Think this was only added in 1.4.0

**Comment by noel-smith** [comment]
LGTM apart from the one minor issue above.

File: python/pyspark/mllib/recommendation.py [hunk 2]
```diff
@@ -36,6 +36,8 @@ class Rating(namedtuple("Rating", ["user", "product", "rating"])):
     (1, 2, 5.0)
     >>> (r[0], r[1], r[2])
     (1, 2, 5.0)
+
+    .. addedversion:: 1.2.0
```
Comment (noel-smith) [inline python/pyspark/mllib/recommendation.py]: Should be `.. versionadded::`

File: python/pyspark/mllib/recommendation.py [hunk 3]
```diff
@@ -111,13 +113,17 @@ class MatrixFactorizationModel(JavaModelWrapper, JavaSaveable, JavaLoader):
     ...     rmtree(path)
     ... except OSError:
     ...     pass
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/recommendation.py]: Should be `.. versionadded::`

File: python/pyspark/mllib/recommendation.py [hunk 4]
```diff
@@ -157,17 +167,25 @@ def recommendProducts(self, user, num):
...
+        """Load a model from the given path"""
         model = cls._load_java(sc, path)
         wrapper = sc._jvm.MatrixFactorizationModelWrapper(model)
         return MatrixFactorizationModel(wrapper)
 
 
 class ALS(object):
+    """Alternating Least Squares matrix factorization
+
+    .. addedversion:: 0.9.0
```
Comment (noel-smith) [inline python/pyspark/mllib/recommendation.py]: Same again - should be `.. versionadded::`

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**